README.md
# bee-popconfirm

[![npm version](https://img.shields.io/npm/v/bee-popconfirm.svg)](https://www.npmjs.com/package/bee-popconfirm)
[![Build Status](https://img.shields.io/travis/tinper-bee/bee-popconfirm/master.svg)](https://travis-ci.org/tinper-bee/bee-popconfirm)
[![Coverage Status](https://coveralls.io/repos/github/tinper-bee/bee-popconfirm/badge.svg?branch=master)](https://coveralls.io/github/tinper-bee/bee-popconfirm?branch=master)


## Browser Support

|![IE](https://raw.github.com/alrra/browser-logos/master/internet-explorer/internet-explorer_48x48.png) | ![Chrome](https://raw.github.com/alrra/browser-logos/master/chrome/chrome_48x48.png) | ![Firefox](https://raw.github.com/alrra/browser-logos/master/firefox/firefox_48x48.png) | ![Opera](https://raw.github.com/alrra/browser-logos/master/opera/opera_48x48.png) | ![Safari](https://raw.github.com/alrra/browser-logos/master/safari/safari_48x48.png)|
| --- | --- | --- | --- | --- |
| IE 9+ ✔ | Chrome 31.0+ ✔ | Firefox 31.0+ ✔ | Opera 30.0+ ✔ | Safari 7.0+ ✔ |


react bee-popconfirm component for tinper-bee

some description...

## 使用方法

```js
const popoverLeft = (
  <Popconfirm id="popover-positioned-left" title="左侧弹出">
    您喜欢使用tinper-bee组件库吗？
  </Popconfirm>
);

class Demo extends Component {
    render () {
        return (
              <OverlayTrigger trigger="click" placement="left" overlay={popoverLeft}>
                <Button colors="primary">向左!</Button>
              </OverlayTrigger>
        )
    }
}
```

#### 样式引入
- 可以使用link引入build目录下Popconfirm.css
```
<link rel="stylesheet" href="./node_modules/bee-popconfirm/build/Popconfirm.css">
```
- 可以在js中import样式
```js
import "./node_modules/bee-popconfirm/src/Popconfirm.scss"
//或是
import "./node_modules/bee-popconfirm/build/Popconfirm.css"
```

## API

|参数|说明|类型|默认值|
|:---|:-----|:----|:------|
|id|标示弹出层|string/number|-|
|placement|弹出位置|top/left/right/bottom|right|
|positionTop|弹出框向上偏移量|string/number|-|
|positionLeft|弹出框向左偏移量|string/number|-|
|arrowOffsetTop|弹出框箭头向上偏移量|string/number|-|
|arrowOffsetLeft|弹出框箭头向左偏移量|string/number|-|
|title|标题|node|-|
|container|标题|node/function|body|


#### 开发调试

```sh
$ npm install -g bee-tools
$ git clone https://github.com/tinper-bee/bee-popconfirm
$ cd bee-popconfirm
$ npm install
$ npm run dev
```


CHANGELOG.md
<a name="0.0.7"></a>
## [0.0.7](https://github.com/tinper-bee/bee-popconfirm/compare/0.0.6...0.0.7) (2017-11-02)



<a name="0.0.6"></a>
## [0.0.6](https://github.com/tinper-bee/bee-popconfirm/compare/0.0.5...0.0.6) (2017-11-02)



<a name="0.0.5"></a>
## [0.0.5](https://github.com/tinper-bee/bee-popconfirm/compare/0.0.4...0.0.5) (2017-11-02)


### Bug Fixes

* **popover:** add .gitignore idea ([aa6d4f5](https://github.com/tinper-bee/bee-popconfirm/commit/aa6d4f5))
* **popover:** test log ([11fef14](https://github.com/tinper-bee/bee-popconfirm/commit/11fef14))



<a name="0.0.4"></a>
## 0.0.4 (2017-11-02)





demo/index.js

import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Button from 'bee-button';


const CARET = <i className="uf uf-chevronarrowdown"></i>;

const CARETUP = <i className="uf uf-chevronarrowup"></i>;


var Demo1 = require("./demolist/Demo1");var Demo2 = require("./demolist/Demo2");var DemoArray = [{"example":<Demo1 />,"title":" 气泡卡片默认样式","code":"/**\n *\n * @title 气泡卡片默认样式\n * @description 支持12个位置\n */\n\nimport {Component} from 'react';\nimport Popover from 'bee-popover';\nimport Button from 'bee-button';\n\n\n\n    constructor(props) {\n        super(props);\n    }\n\n    render() {\n        let content = (\n            <div>\n                <h3>消息</h3>\n                <ul>\n                    <li>您的服务器已宕机，请处理。</li>\n                    <li>您的手机已停机，请续费。</li>\n                    <li>你的工资已到账，请查收。</li>\n                </ul>\n            </div>\n        )\n        return (\n            <div style={{position: 'relative', height: 320, marginTop: 100, marginLeft: 250}}>\n                <Popover\n                    placement=\"leftTop\"\n                    content={content}\n                    trigger=\"hover\"\n                    id=\"leftTop\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 5,\n                            left: 0\n                        }}>左上</Button>\n                </Popover>\n                <Popover\n                    placement=\"left\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"left\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 55,\n                            left: 0\n                        }}>左\n                    </Button>\n                </Popover>\n                <Popover\n                    placement=\"leftBottom\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"leftBottom\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 110,\n                            left: 0\n                        }}>左下</Button>\n                </Popover>\n                <Popover\n                    placement=\"topLeft\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"topLeft\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 0,\n                            left: 100\n                        }}>上左\n                    </Button>\n                </Popover>\n                <Popover\n                    placement=\"top\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"top\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 0,\n                            left: 200\n                        }}>上\n                    </Button>\n                </Popover>\n                <Popover\n                    placement=\"topRight\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"topRight\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 0,\n                            left: 300\n                        }}>\n                        上右\n                    </Button>\n                </Popover>\n                <Popover\n                    placement=\"rightTop\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"rightTop\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 5,\n                            left: 400\n                        }}>右上\n                    </Button>\n                </Popover>\n                <Popover\n                    placement=\"right\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"right\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 55,\n                            left: 400\n                        }}>右</Button>\n                </Popover>\n                <Popover\n                    placement=\"rightBottom\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"rightBottom\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 110,\n                            left: 400\n                        }}>右下</Button>\n                </Popover>\n                <Popover\n                    placement=\"bottomLeft\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"bottomLeft\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 115,\n                            left: 100\n                        }}>下左</Button>\n                </Popover>\n                <Popover\n                    placement=\"bottom\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"bottom\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 115,\n                            left: 200\n                        }}>上</Button>\n                </Popover>\n                <Popover\n                    placement=\"bottomRight\"\n                    content={content}\n                    trigger=\"click\"\n                    id=\"bottomRight\"\n                >\n                    <Button\n                        colors=\"primary\"\n                        style={{\n                            position: 'absolute',\n                            top: 115,\n                            left: 300\n                        }}>下右</Button>\n                </Popover>\n            </div>\n        )\n    }\n}\n","desc":" 支持12个位置"},{"example":<Demo2 />,"title":" 受控制的气泡卡片","code":"/**\r\n *\r\n * @title 受控制的气泡卡片\r\n * @description 通过设置show，来控制气泡卡片的显示和隐藏\r\n */\r\n\r\nimport {Component} from 'react';\r\nimport Popover from 'bee-popover';\r\nimport Button from 'bee-button';\r\n\r\n\r\n\r\n    constructor(props) {\r\n        super(props);\r\n        this.state = {\r\n            show: false\r\n        }\r\n    }\r\n\r\n    handleClose = () => {\r\n        this.setState({\r\n            show: false\r\n        })\r\n    }\r\n\r\n    show = () => {\r\n        this.setState({\r\n            show: true\r\n        })\r\n    }\r\n\r\n    render() {\r\n        let content = (\r\n            <div>\r\n                <p>请确认您的包裹已签收！</p>\r\n                <div>\r\n                    <Button\r\n                        colors=\"primary\"\r\n                        onClick={ this.handleClose }\r\n                        size=\"sm\">\r\n                        关闭\r\n                    </Button>\r\n                </div>\r\n            </div>\r\n        )\r\n        return (\r\n            <div>\r\n                <Popover\r\n                    placement=\"right\"\r\n                    content={content}\r\n                    show={this.state.show}\r\n                    id=\"demo2\"\r\n                >\r\n                    <Button\r\n                        colors=\"primary\"\r\n                        onClick={ this.show }\r\n                        >确认按钮</Button>\r\n                </Popover>\r\n            </div>\r\n        )\r\n    }\r\n}\r\n","desc":" 通过设置show，来控制气泡卡片的显示和隐藏"}]


class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.setState({ open: !this.state.open })
    }

    render () {
        const { title, example, code, desc  } = this.props;
        let caret = this.state.open ? CARETUP : CARET;
        let text = this.state.open ? "隐藏代码" : "查看代码";

        const footer = (
            <Button shape="block" onClick={ this.handleClick }>
                { caret }
                { text }
            </Button>
        );

        
        return (
            <Col md={12}>
                <h3>{ title }</h3>
                <p>{ desc }</p>
                <Panel collapsible headerContent expanded={ this.state.open } colors='bordered' header={ example } footer={footer} footerStyle = {{padding: 0, borderColor: "transparent"}}>
                    <pre><code className="hljs javascript">{ code }</code></pre>
                </Panel>
            </Col>
        )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
                <Row>
                    {DemoArray.map((child,index) => {

                        return (
                            <Demo example= {child.example} title= {child.title} code= {child.code} desc= {child.desc} key= {index}/>
                        )

                    })}
                </Row>
        )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));


src/Content.js
import React from 'react';
import isRequiredForA11y from 'tinper-bee-core/lib/isRequiredForA11y';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import OverlayTrigger from 'bee-overlay/build/OverlayTrigger';


const propTypes = {
    /**
     * An html id attribute, necessary for accessibility
     * @type {string}
     * @required
     */
    id: isRequiredForA11y(PropTypes.oneOfType([
        PropTypes.string, PropTypes.number,
    ])),

    /**
     * Sets the direction the Popover is positioned towards.
     */
    placement: PropTypes.oneOf(['top', 'right', 'bottom', 'left','topLeft', 'rightTop', 'bottomLeft', 'leftTop','topRight', 'rightBottom', 'bottomRight', 'leftBottom']),

    /**
     * The "top" position value for the Popover.
     */
    positionTop: PropTypes.oneOfType([
        PropTypes.number, PropTypes.string,
    ]),
    /**
     * The "left" position value for the Popover.
     */
    positionLeft: PropTypes.oneOfType([
        PropTypes.number, PropTypes.string,
    ]),

    /**
     * The "top" position value for the Popover arrow.
     */
    arrowOffsetTop: PropTypes.oneOfType([
        PropTypes.number, PropTypes.string,
    ]),
    /**
     * The "left" position value for the Popover arrow.
     */
    arrowOffsetLeft: PropTypes.oneOfType([
        PropTypes.number, PropTypes.string,
    ]),
};

const defaultProps = {
    placement: 'right',
    clsPrefix: 'u-popover'
};

const PLACECLASS = {
    right: 'right',
    top: 'top',
    bottom: 'bottom',
    left: 'left',
    rightTop: 'right-top',
    rightBottom: 'right-bottom',
    leftTop: 'left-top',
    leftBottom: 'left-bottom',
    topRight: 'top-right',
    topLeft: 'top-left',
    bottomLeft: 'bottom-left',
    bottomRight: 'bottom-right'
};

class Content extends React.Component {
    render() {
        const {
            placement,
            positionTop,
            positionLeft,
            arrowOffsetTop,
            arrowOffsetLeft,
            clsPrefix,
            className,
            style,
            id,
            children,
            trigger,
            ...others
        } = this.props;

        const classes = {
            [`${clsPrefix}`]: true,
            [PLACECLASS[placement]]: true,
        };

        const outerStyle = {
            display: 'block',
            top: positionTop,
            left: positionLeft,
            ...style,
        };

        const arrowStyle = {
            top: arrowOffsetTop,
            left: arrowOffsetLeft,
        };

        return (
            <div
                role="tooltip"
                id={id}
                className={classnames(className, classes)}
                style={outerStyle}
                {...others}
            >
                <div className="arrow" style={arrowStyle}/>

                <div className={classnames(`${clsPrefix}-content`)}>
                    { children }
                </div>
            </div>
        );
    }
}

Content.propTypes = propTypes;
Content.defaultProps = defaultProps;

export default Content;


src/Popover.js
import React, { Component, cloneElement } from 'react';
import ReactDOM from 'react-dom';
import createChainedFunction from 'tinper-bee-core/lib/createChainedFunction';
import splitComponentProps from 'tinper-bee-core/lib/splitComponent';
import PropTypes from 'prop-types';
import Overlay from 'bee-overlay/build/Overlay';
import Portal from 'bee-overlay/build/Portal';
import Content from './Content';
import contains from 'dom-helpers/query/contains';

//TODO: 当多个Popover在一个组件内时，显示一个会触发多个渲染。见demo1.

const isReact16 = ReactDOM.createPortal !== undefined;

const triggerType = PropTypes.oneOf(['click', 'hover', 'focus']);

/**
 * 检查值是属于这个值，还是等于这个值
 *
 * @param {string} one
 * @param {string|array} of
 * @returns {boolean}
 */
function isOneOf(one, of) {
    if (Array.isArray(of)) {
        return of.indexOf(one) >= 0;
    }
    return one === of;
}

const propTypes = {
    ...Overlay.propTypes,


    // FIXME: This should be `defaultShow`.
    /**
     * 覆盖的初始可见性状态。对于更细微的可见性控制，请考虑直接使用覆盖组件。
     */
    defaultOverlayShown: PropTypes.bool,

    /**
     * 要覆盖在目标旁边的元素或文本。
     */
    content: PropTypes.node.isRequired,
    /**
     * 显示和隐藏覆盖一旦触发的毫秒延迟量
     */
    delay: PropTypes.number,
    /**
     * 触发后显示叠加层之前的延迟毫秒
     */
    delayShow: PropTypes.number,
    /**
     * 触发后隐藏叠加层的延迟毫秒
     */
    delayHide: PropTypes.number,

    /**
     * @private
     */
    onClick: PropTypes.func,
    onClose: PropTypes.func,
    onCancel: PropTypes.func,


    // Overridden props from `<Overlay>`.
    /**
     * @private
     */
    target: PropTypes.oneOf([null]),
    /**
     * @private
     */
    onHide: PropTypes.oneOf([null]),
    /**
     * @private
     */
    show: PropTypes.bool,

    trigger: PropTypes.oneOfType([
        triggerType, PropTypes.arrayOf(triggerType),
    ]),
    /**
     * @private
     */
    onBlur: PropTypes.func,
    /**
     * @private
     */
    onFocus: PropTypes.func,
    /**
     * @private
     */
    onMouseOut: PropTypes.func,
    /**
     * @private
     */
    onMouseOver: PropTypes.func,
};


const defaultProps = {
    placement: 'right',
    clsPrefix: 'u-popover',
    rootClose: true,
    defaultOverlayShown: false,
};

class Popover extends Component{
    constructor(props, context) {
        super(props, context);

        this._mountNode = null;

        this.state = {
            show: props.defaultOverlayShown,
        };

        this.handleMouseOver = e => (
            this.handleMouseOverOut(this.handleDelayedShow, e)
        );
        this.handleMouseOut = e => (
            this.handleMouseOverOut(this.handleDelayedHide, e)
        );
    }

    componentDidMount() {
        this._mountNode = document.createElement('div');
        !isReact16 && this.renderOverlay();
    }

    componentWillReceiveProps(nextProps) {
        if(nextProps.hasOwnProperty('show')){
            if(nextProps.show){
                this.handleToggle();
            }else{
                this.handleHide();
            }
        }
    }

    componentDidUpdate() {
        !isReact16 && this.renderOverlay();
    }

    componentWillUnmount() {
        !isReact16 && ReactDOM.unmountComponentAtNode(this._mountNode);
        this._mountNode = null;

    }

    handleToggle = () => {

        if (!this.state.show) {
            this.show();
        }else{
            this.hide();
        }
    }

    handleDelayedShow = () => {
        if (this._hoverHideDelay != null) {
            clearTimeout(this._hoverHideDelay);
            this._hoverHideDelay = null;
            return;
        }

        if (this.state.show || this._hoverShowDelay != null) {
            return;
        }

        const delay = this.props.delayShow != null ?
            this.props.delayShow : this.props.delay;

        if (!delay) {
            this.show();
            return;
        }

        this._hoverShowDelay = setTimeout(() => {
            this._hoverShowDelay = null;
            this.show();
        }, delay);
    }

    handleDelayedHide = () => {
        if (this._hoverShowDelay != null) {
            clearTimeout(this._hoverShowDelay);
            this._hoverShowDelay = null;
            return;
        }

        if (!this.state.show || this._hoverHideDelay != null) {
            return;
        }

        const delay = this.props.delayHide != null ?
            this.props.delayHide : this.props.delay;

        if (!delay) {
            this.hide();
            return;
        }

        this._hoverHideDelay = setTimeout(() => {
            this._hoverHideDelay = null;
            this.hide();
        }, delay);
    }

    // 简单实现mouseEnter和mouseLeave。
    // React的内置版本是有问题的：https://github.com/facebook/react/issues/4251
    //在触发器被禁用的情况下，mouseOut / Over可能导致闪烁
    //从一个子元素移动到另一个子元素。
    handleMouseOverOut = (handler, e) => {
        const target = e.currentTarget;
        const related = e.relatedTarget || e.nativeEvent.toElement;

        if (!related || related !== target && !contains(target, related)) {
            handler(e);
        }
    }


    handleHide = () => {
        this.hide();
    }

    show = () => {
        this.setState({ show: true });
    }

    hide = () => {
        this.setState({ show: false });
    }

    makeOverlay = (overlay, props) => {
        return (
            <Overlay
                {...props}
                show={this.state.show}
                onHide={this.handleHide}
                target={this}
            >
                {overlay}
            </Overlay>
        );
    }

    renderOverlay = () => {
        ReactDOM.unstable_renderSubtreeIntoContainer(
            this, this._overlay, this._mountNode
        );
    }

    render() {
        const {
            content,
            children,
            onClick,
            trigger,
            onBlur,
            onFocus,
            onMouseOut,
            onMouseOver,
            ...props
        } = this.props;

        delete props.delay;
        delete props.delayShow;
        delete props.delayHide;
        delete props.defaultOverlayShown;

        const [overlayProps, confirmProps] = splitComponentProps(props, Overlay);

        const child = React.Children.only(children);
        const childProps = child.props;

        let overlay = (
            <Content placement={ props.placement } {...confirmProps} >
                {content}
            </Content>
        );

        const triggerProps = {
            'aria-describedby': overlay.props.id
        };

        // FIXME: 这里用于传递这个组件上的处理程序的逻辑是不一致的。我们不应该通过任何这些道具。

        triggerProps.onClick = createChainedFunction(childProps.onClick, onClick);

        if (isOneOf('click', trigger)) {
            triggerProps.onClick = createChainedFunction(
                triggerProps.onClick, this.handleToggle
            );
        }

        if (isOneOf('hover', trigger)) {

            triggerProps.onMouseOver = createChainedFunction(
                childProps.onMouseOver, onMouseOver, this.handleMouseOver
            );
            triggerProps.onMouseOut = createChainedFunction(
                childProps.onMouseOut, onMouseOut, this.handleMouseOut
            );
        }

        if (isOneOf('focus', trigger)) {
            triggerProps.onFocus = createChainedFunction(
                childProps.onFocus, onFocus, this.handleDelayedShow
            );
            triggerProps.onBlur = createChainedFunction(
                childProps.onBlur, onBlur, this.handleDelayedHide
            );
        }


        this._overlay = this.makeOverlay(overlay, overlayProps);

        if (!isReact16) {
            return cloneElement(child, triggerProps);
        }
        triggerProps.key = 'overlay';

        let portal = (
            <Portal
                key="portal"
                container={props.container}>
                { this._overlay }
            </Portal>
        )


        return [
            cloneElement(child, triggerProps),
            portal
        ]
    }

}

Popover.propTypes = propTypes;
Popover.defaultProps = defaultProps;

export default Popover;


demo/demolist/Demo2.js
/**
 *
 * @title 受控制的气泡卡片
 * @description 通过设置show，来控制气泡卡片的显示和隐藏
 */

import {Component} from 'react';
import Popover from '../../src';
import Button from 'bee-button';


export default class Demo2 extends Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false
        }
    }

    handleClose = () => {
        this.setState({
            show: false
        })
    }

    show = () => {
        this.setState({
            show: true
        })
    }

    render() {
        let content = (
            <div>
                <p>请确认您的包裹已签收！</p>
                <div>
                    <Button
                        colors="primary"
                        onClick={ this.handleClose }
                        size="sm">
                        关闭
                    </Button>
                </div>
            </div>
        )
        return (
            <div>
                <Popover
                    placement="right"
                    content={content}
                    show={this.state.show}
                    id="demo2"
                >
                    <Button
                        colors="primary"
                        onClick={ this.show }
                        >确认按钮</Button>
                </Popover>
            </div>
        )
    }
}


demo/index-demo-base.js

import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Button from 'bee-button';


const CARET = <i className="uf uf-chevronarrowdown"></i>;

const CARETUP = <i className="uf uf-chevronarrowup"></i>;


{demolist}

class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.setState({ open: !this.state.open })
    }

    render () {
        const { title, example, code, desc  } = this.props;
        let caret = this.state.open ? CARETUP : CARET;
        let text = this.state.open ? "隐藏代码" : "查看代码";

        const footer = (
            <Button shape="block" onClick={ this.handleClick }>
                { caret }
                { text }
            </Button>
        );

        
        return (
            <Col md={12}>
                <h3>{ title }</h3>
                <p>{ desc }</p>
                <Panel collapsible headerContent expanded={ this.state.open } colors='bordered' header={ example } footer={footer} footerStyle = {{padding: 0, borderColor: "transparent"}}>
                    <pre><code className="hljs javascript">{ code }</code></pre>
                </Panel>
            </Col>
        )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
                <Row>
                    {DemoArray.map((child,index) => {

                        return (
                            <Demo example= {child.example} title= {child.title} code= {child.code} desc= {child.desc} key= {index}/>
                        )

                    })}
                </Row>
        )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));


demo/demolist/Demo1.js
/**
 *
 * @title 气泡卡片默认样式
 * @description 支持12个位置
 */

import {Component} from 'react';
import Popover from '../../src';
import Button from 'bee-button';


export default class Demo1 extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        let content = (
            <div>
                <h3>消息</h3>
                <ul>
                    <li>您的服务器已宕机，请处理。</li>
                    <li>您的手机已停机，请续费。</li>
                    <li>你的工资已到账，请查收。</li>
                </ul>
            </div>
        )
        return (
            <div style={{position: 'relative', height: 320, marginTop: 100, marginLeft: 250}}>
                <Popover
                    placement="leftTop"
                    content={content}
                    trigger="hover"
                    id="leftTop"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 5,
                            left: 0
                        }}>左上</Button>
                </Popover>
                <Popover
                    placement="left"
                    content={content}
                    trigger="click"
                    id="left"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 55,
                            left: 0
                        }}>左
                    </Button>
                </Popover>
                <Popover
                    placement="leftBottom"
                    content={content}
                    trigger="click"
                    id="leftBottom"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 110,
                            left: 0
                        }}>左下</Button>
                </Popover>
                <Popover
                    placement="topLeft"
                    content={content}
                    trigger="click"
                    id="topLeft"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 0,
                            left: 100
                        }}>上左
                    </Button>
                </Popover>
                <Popover
                    placement="top"
                    content={content}
                    trigger="click"
                    id="top"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 0,
                            left: 200
                        }}>上
                    </Button>
                </Popover>
                <Popover
                    placement="topRight"
                    content={content}
                    trigger="click"
                    id="topRight"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 0,
                            left: 300
                        }}>
                        上右
                    </Button>
                </Popover>
                <Popover
                    placement="rightTop"
                    content={content}
                    trigger="click"
                    id="rightTop"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 5,
                            left: 400
                        }}>右上
                    </Button>
                </Popover>
                <Popover
                    placement="right"
                    content={content}
                    trigger="click"
                    id="right"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 55,
                            left: 400
                        }}>右</Button>
                </Popover>
                <Popover
                    placement="rightBottom"
                    content={content}
                    trigger="click"
                    id="rightBottom"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 110,
                            left: 400
                        }}>右下</Button>
                </Popover>
                <Popover
                    placement="bottomLeft"
                    content={content}
                    trigger="click"
                    id="bottomLeft"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 115,
                            left: 100
                        }}>下左</Button>
                </Popover>
                <Popover
                    placement="bottom"
                    content={content}
                    trigger="click"
                    id="bottom"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 115,
                            left: 200
                        }}>上</Button>
                </Popover>
                <Popover
                    placement="bottomRight"
                    content={content}
                    trigger="click"
                    id="bottomRight"
                >
                    <Button
                        colors="primary"
                        style={{
                            position: 'absolute',
                            top: 115,
                            left: 300
                        }}>下右</Button>
                </Popover>
            </div>
        )
    }
}
